clients/src/test/java/org/apache/kafka/clients/admin/AdminClientUnitTestEnv.java

```diff
@@ -82,7 +87,15 @@ public void update(Time time, MockClient.MetadataUpdate update) {
         });
 
         metadataManager.update(cluster, time.milliseconds());
+        adminClient = null;
```

Comment:
@stanislavkozlovski Couldn't we add a new constructor with a boolean parameter to decide whether `adminClient` should be created here? So all tests except `testUnreachableBootstrapServer` don't need to remember to invoke `initiate()`. And `testUnreachableBootstrapServer` will be explicitly not creating the admin client, so it will be more obvious that it needs to create it separately. Perhaps that would be less error-prone for the future?

Reply:
Sounds good. I updated the javadoc to describe that as well